std/io.zig

```diff
@@ -290,7 +290,7 @@ pub fn readFileAllocAligned(allocator: *mem.Allocator, path: []const u8, comptim
     var file = try File.openRead(path);
     defer file.close();
 
-    const size = try file.getEndPos();
+    const size = @intCast(usize, try file.getEndPos());
```

Comment:
Counter-proposal:

```zig
     const size = try std.math.cast(usize, try file.getEndPos()); 
```

This adds `error.Overflow` as one of the possible errors of `std.io.readFileAllocAligned`, which I think is reasonable, since on 32 bit architectures, the file size could be larger than what fits in `usize`.

In the future I plan to add a mandatory `max_size: usize` argument to `readFileAlloc` and `readFileAllocAligned` which will return `error.FileTooBig` if the file size exceeds the value. Feel free to do that now if you wish.

Reply:
Done.

> In the future I plan to add a mandatory max_size: usize argument to readFileAlloc and readFileAllocAligned which will return error.FileTooBig if the file size exceeds the value. Feel free to do that now if you wish.

I think that making it optional makes the API more ergonomic, having to fish up  `math.maxInt(usize)` every time you just want to read a file gets old pretty fast IMO.

Reply:
Well that's the thing - is `math.maxInt(usize)` really what people would want? That seems too big. If a file could be that large, probably reading it all at once into memory is the wrong idea. If it's a "config file" or similar, then setting something reasonable like a 10 MiB maximum would make more sense. If it's a user supplied file then certainly a size limit would be in order. The idea is that `math.maxInt(usize)` does feel a *little* awkward, and that would give people enough pause to consider a proper limit for this function call.

Anyway that's a separate issue.